hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/AccountStore.java

```diff
@@ -99,6 +99,14 @@ public Account loadContract(final Id id) {
         return loadEntityOrFailWith(id, null, INVALID_CONTRACT_ID, CONTRACT_DELETED);
     }
 
+    public boolean isContractUsable(final Id id) {
+        // check existence
+        // check expired
+        // check deleted
```

Comment:
for expired and deleted contracts we should confirm what the expected behavior is. There is no such concept in Ethereum so this would be purely a Hedera decision.